examples/ASDKgram/Sample/PhotoCellNode.m

```diff
@@ -0,0 +1,188 @@
+#import "PINButton+PINRemoteImage.h"
+
+#define DEBUG_PHOTOCELL_LAYOUT  0
+
+#define HEADER_HEIGHT           50
+#define USER_IMAGE_HEIGHT       30
+#define HORIZONTAL_BUFFER       10
+#define VERTICAL_BUFFER         5
+#define FONT_SIZE               14
+
+@implementation PhotoCellNode
+{
+  PhotoModel          *_photoModel;
+  CommentsNode        *_photoCommentsView;
+  ASNetworkImageNode  *_userAvatarImageView;
+  ASNetworkImageNode  *_photoImageView;
+  ASTextNode          *_userNameLabel;
+  ASTextNode          *_photoLocationLabel;
+  ASTextNode          *_photoTimeIntervalSincePostLabel;
+  ASTextNode          *_photoLikesLabel;
+  ASTextNode          *_photoDescriptionLabel;
+}
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithPhotoObject:(PhotoModel *)photo;
+{
+  self = [super init];
+  
+  if (self) {
+    
+    _photoModel              = photo;
+    
+    _userAvatarImageView     = [[ASNetworkImageNode alloc] init];
+    _userAvatarImageView.URL = photo.ownerUserProfile.userPicURL;   // FIXME: make round
+    
+    // FIXME: autocomplete for this line seems broken
+    [_userAvatarImageView setImageModificationBlock:^UIImage *(UIImage *image) {
+      CGSize profileImageSize = CGSizeMake(USER_IMAGE_HEIGHT, USER_IMAGE_HEIGHT);
+      return [image makeCircularImageWithSize:profileImageSize];
+    }];
+
+    _photoImageView          = [[ASNetworkImageNode alloc] init];
+    _photoImageView.URL      = photo.URL;
+    _photoImageView.layerBacked = YES;
+    
+    _userNameLabel                  = [[ASTextNode alloc] init];
+    _userNameLabel.attributedString = [photo.ownerUserProfile usernameAttributedStringWithFontSize:FONT_SIZE];
+    
+    _photoLocationLabel      = [[ASTextNode alloc] init];
+    _photoLocationLabel.maximumNumberOfLines = 1;
+    [photo.location reverseGeocodedLocationWithCompletionBlock:^(LocationModel *locationModel) {
+      
+      // check and make sure this is still relevant for this cell (and not an old cell)
+      // make sure to use _photoModel instance variable as photo may change when cell is reused,
+      // where as local variable will never change
+      if (locationModel == _photoModel.location) {
+        _photoLocationLabel.attributedString = [photo locationAttributedStringWithFontSize:FONT_SIZE];
+        [self setNeedsLayout];
+      }
+    }];
+    
+    _photoTimeIntervalSincePostLabel                  = [[ASTextNode alloc] init];
+    _photoTimeIntervalSincePostLabel.layerBacked      = YES;
+    _photoTimeIntervalSincePostLabel.attributedString = [photo uploadDateAttributedStringWithFontSize:FONT_SIZE];
+    
+    _photoLikesLabel                        = [[ASTextNode alloc] init];
+    _photoLikesLabel.layerBacked            = YES;
+    _photoLikesLabel.attributedString       = [photo likesAttributedStringWithFontSize:FONT_SIZE];
+    
+    _photoDescriptionLabel                  = [[ASTextNode alloc] init];
+    _photoDescriptionLabel.layerBacked      = YES;
```

Comment:
You might have enough text nodes to warrant a -createTextNodeWithString: method, esp because they all set layerBacked.